presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/optimization/TestJdbcComputePushdown.java

```diff
@@ -0,0 +1,143 @@
+import com.facebook.presto.spi.plan.PlanNodeIdAllocator;
+import com.facebook.presto.spi.plan.TableScanNode;
+import com.facebook.presto.spi.predicate.TupleDomain;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.assertions.MatchResult;
+import com.facebook.presto.sql.planner.assertions.Matcher;
+import com.facebook.presto.sql.planner.assertions.PlanAssert;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.sql.planner.assertions.SymbolAliases;
+import com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder;
+import com.facebook.presto.testing.TestingConnectorSession;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.spi.relation.LogicalRowExpressions.TRUE_CONSTANT;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.node;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+public class TestJdbcComputePushdown
+{
+    private static final Metadata METADATA = MetadataManager.createTestMetadataManager();
+    private static final PlanBuilder PLAN_BUILDER = new PlanBuilder(TEST_SESSION, new PlanNodeIdAllocator(), METADATA);
+    private static final PlanNodeIdAllocator ID_ALLOCATOR = new PlanNodeIdAllocator();
+
+    private final JdbcComputePushdown jdbcComputePushdown;
+
+    public TestJdbcComputePushdown()
+    {
+        this.jdbcComputePushdown = new JdbcComputePushdown(null, null, null, null);
+    }
+
+    @Test
+    public void testJdbcComputePushdownIsNoop()
+    {
+        JdbcTableHandle jdbcTableHandle = new JdbcTableHandle("cat1", new SchemaTableName("schema", "table"), null, null, "table");
+        JdbcTableLayoutHandle jdbcTableLayoutHandle = new JdbcTableLayoutHandle(jdbcTableHandle, TupleDomain.none());
+        TableHandle tableHandle = new TableHandle(new ConnectorId("Jdbc"), jdbcTableHandle, new ConnectorTransactionHandle() {}, Optional.of(jdbcTableLayoutHandle));
+        PlanNode original = filter(tableScan(tableHandle, "a", "b"), TRUE_CONSTANT);
+
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+        PlanNode actual = this.jdbcComputePushdown.optimize(original, session, null, ID_ALLOCATOR);
+        assertPlanMatch(actual, PlanMatchPattern.filter(
+                TRUE_CONSTANT.toString(), JdbcTableScanMatcher.jdbcTableScanPattern(jdbcTableLayoutHandle)));
```

Comment:
Lmk if this is the right way to write tests for this. I looked at the example in TestConnectorOptimization/TestScubaFilterPushdown etc. but there were slight changes due to the way Jdbc layout handle and table handles work.